CMakeLists.txt

```diff
@@ -21,18 +21,24 @@ include(plugin.cmake OPTIONAL)
 
 include(ZeekPlugin)
 
-zeek_add_plugin(
-    Zeek
-    PerfSupport
-    SOURCES
-    src/Plugin.cc
-    src/Trampoline.S
-    DIST_FILES
-    README.md
-    CHANGES
-    COPYING
-    VERSION
-)
+zeek_plugin_begin(Zeek PerfSupport)
+zeek_plugin_cc(src/Plugin.cc src/Trampoline.S)
+zeek_plugin_dist_files(README.md CHANGES COPYING VERSION)
+zeek_plugin_end()
```

Comment:
What's the reason for switching from `zeek_add_plugin` to the "exploded" API? The former seems subjectively "better" to me since sidesteps the subtle state keeping of the exploded API. We also don't seem to inject anything, and I do not understand why we would need it.

Reply:
Because `zeek_add_plugin()` is broken in 7.0 and 7.1 for builtin plugins, but the old API still works. The commit mentions this.


https://github.com/zeek/cmake/pull/125

Reply:
Since this is something which must not be cleaned up, can you please add a comment for this here?

Reply:
Hmm, there's the whole line below which intended to be a reminder, but yes, I can restructure this to be more obvious.

Reply:
Cleaned it up, there was nothing in the commit message, acutally :sweat: